providers/linux/host_fqdn_nocgo_linux.go

```diff
@@ -0,0 +1,28 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build !cgo
+
+package linux
+
+func fqdn() (string, error) {
+	return "", nil
+}
+
+func domainname() (string, error) {
```

Comment:
I think the NIS domain can be read from `/proc/sys/kernel/domainname`. Then you wouldn't need separate CGO and non-CGO implementations. And probably the docker based integration testing could be removed since you could simulate various hostname and domainname via the filesystem (probably like  `newLinuxSystem("testdata/my_test_proc")`).

https://www.kernel.org/doc/html/latest/admin-guide/sysctl/kernel.html#domainname-hostname

Reply:
Do we really want to bring the NIS domain into FQDN?  As I understand it, FQDN is part of DNS not NIS.  [Gnu libc](https://www.gnu.org/software/libc/manual/html_node/Host-Identification.html) and [RFC1594](https://datatracker.ietf.org/doc/html/rfc1594#section-5) seem to agree.

Reply:
I have the same question. I was alluding to this in my other comment at https://github.com/elastic/go-sysinfo/pull/144#discussion_r1088114867. Here I was mainly focused on the code as it was without debating DNS vs NIS. I haven't looked at the latest changes to see how this has evolved, but I still think we need to clear on what we are reporting.